### ISSUE TITLE
Add attribute from more than 1 depth

### ISSUE THREAD

**Issue by Albvadi** [laravel-json-api/laravel#116]
Hi, 
I´m trying to add attributes from a relationship from more than one depth. Is it possible?

Example:
I have the following schema:
```
Country
  |
  State
    |
    City
```
```php
class Country extends Model
{
    public function states()
    {
        return $this->hasMany(States::class);
    }
}

class State extends Model
{
    public function country()
    {
        return $this->belongsTo(Country::class);
    }

    public function cities()
    {
        return $this->hasMany(City::class);
    }
}

class Recurso extends Model
{
    public function states()
    {
        return $this->belongsTo(State::class);
    }
}
```
In the last schema `City` I want the name of the parents, `State` and `Country`.
I can get the last parent with [Attributes from Related Models](https://laraveljsonapi.io/docs/1.0/schemas/attributes.html#attributes-from-related-models) but not the very first parent:

CitySchema
```php
class CitySchema extends Schema
{
    protected int $maxDepth = 4;

    public function fields(): array
    {
        return [
            ID::make(),
            Str::make('name'),            
            Map::make('geo', [
                Str::make('state_name')->on('states'), // The correct name of the state
                Str::make('country_name')->on('states.country'), // null
            ]),
        ];
    }
}
```
Is there any way to get the attribute I need?

Thank you!

**Comment by lindyhopchris** [comment]
We don't support this to a depth of more than one. This is because it'll get far too complex to handle to a bigger depth, plus the functionality is designed to work with Eloquent's relationship `withDefault()` method - so that we can successfully create the related model when the parent model is also being created.

If you don't need to ever fill these attributes - i.e. you're just serializing them - then you should use a `Resource` class. The `Resource` class is optional, but can be used for complex serialization... which covers your scenario. Resources are documented here:
https://laraveljsonapi.io/docs/1.0/resources/

I'm going to close this issue as I've answered the question but feel free to ask a follow up question if needed.

**Comment by Albvadi** [comment]
Following your recomendation, I added a CityResource and It has been as simple as adding:
```php
'geo' => [
    'city' => $this->name,
    'state' => $this->state->name,
    'country' => $this->state->country->name,
],
```
But another question arises. I have done this without adding the relations in the `relationships` function. And since it works like that, what is it worth exactly if the fields are already available without adding them?

Thank you for your quick response!

**Comment by lindyhopchris** [comment]
Hey. Not entirely sure I understand your question. Can you rephrase?

**Comment by Albvadi** [comment]
Forget it, I have answered myself by testing the Resources.

Again, thank you very much for the support!
